stdlib/cache/src/main/ballerina/src/cache/cache.bal

```diff
@@ -14,336 +14,308 @@
 // specific language governing permissions and limitations
 // under the License.
 
-import ballerina/system;
 import ballerina/task;
 import ballerina/time;
 
-# Cache cleanup task invoking interval in ms.
-const int CACHE_CLEANUP_INTERVAL = 5000;
+# The `LRU` eviction algorithm.
+public const LRU = "Least Recently Used";
+# The `FIFO` eviction algorithm.
+public const FIFO = "First In First Out";
 
-# Map which stores all of the caches.
-map<Cache> cacheMap = {};
+# The collection of eviction algorithms.
+public type EvictionPolicy LRU|FIFO;
 
-task:TimerConfiguration cacheCleanupTimerConfiguration = {
-    intervalInMillis: CACHE_CLEANUP_INTERVAL,
-    initialDelayInMillis: CACHE_CLEANUP_INTERVAL
-};
-
-task:Scheduler cacheCleanupTimer = new(cacheCleanupTimerConfiguration);
-
-boolean timerStarted = false;
-
-# Represents a cache entry.
+# Represents cache configuration.
 #
-# + value - cache value
-# + lastAccessedTime - last accessed time in ms of this value which is used to remove LRU cached values
+# + capacity - Maximum number of entries allowed
+# + evictionPolicy - The policy which defines the cache eviction algorithm
+# + evictionFactor - The factor which the entries will be evicted once the cache full
+# + defaultMaxAgeInSeconds - The default value in seconds which all the cache entries are valid.
+#                            '-1' means, the entries are valid forever.
+#                            This will be overwritten by the the `maxAge` property set when inserting item to cache
+# + timerIntervalInSeconds - Interval of the timer task which clean up the cache
+public type CacheConfig record {|
+    int capacity = 100;
+    EvictionPolicy evictionPolicy = LRU;
+    float evictionFactor = 0.25;
+    int defaultMaxAgeInSeconds = -1;
+    int timerIntervalInSeconds?;
```

Comment:
Shall we rename this to something which conveys what this time interval is about?

Reply:
@pubudu91 How about `cleanUpTimerIntervalInSeconds`? Then the length of the variable name becomes larger. Any suggestions?

Reply:
How about just `cleanupIntervalInSeconds`?

Reply:
That's better. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/21308/commits/97722a464ecd43b8440046e804458ff2209c135b. Thanks for the suggestion.